File: modules/zig/src/main/java/com/falsepattern/zigbrains/zig/psi/mixins/ZigStringLiteralMixin.java
package com.falsepattern.zigbrains.zig.psi.mixins;

import com.intellij.openapi.util.TextRange;
import com.intellij.psi.PsiLanguageInjectionHost;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface ZigStringLiteralMixin extends PsiLanguageInjectionHost {
    boolean isMultiLine();
    @NotNull List<TextRange> getContentRanges();
}

File: modules/zig/src/main/kotlin/com/falsepattern/zigbrains/zig/psi/mixins/ZigStringLiteralMixin.kt
package com.falsepattern.zigbrains.zig.psi.mixins

import com.intellij.openapi.util.TextRange
import com.intellij.psi.PsiLanguageInjectionHost

interface ZigStringLiteralMixin: PsiLanguageInjectionHost {
    val isMultiline: Boolean
    val contentRanges: List<TextRange>
}